File: Java/opentraceability/src/main/java/opentraceability/models/events/EPCISQueryDocument.java
package opentraceability.models.events;

public class EPCISQueryDocument extends EPCISBaseDocument {
    public String QueryName = "";
    public String SubscriptionID = "";

    public EPCISDocument ToEPCISDocument() {
        EPCISDocument document = new EPCISDocument();

        // get all properties from EPCISBaseDocument
        java.lang.reflect.Field[] props = EPCISBaseDocument.class.getDeclaredFields();

        // iterate over properties and copy their values to document
        for (java.lang.reflect.Field p : props) {
            p.setAccessible(true);
            try {
                Object v = p.get(this);
                p.set(document, v);
            } catch (IllegalAccessException e) {
                e.printStackTrace();
            }
        }

        return document;
    }
}

File: Kotlin/OpenTraceability/src/main/kotlin/opentraceability/models/events/epcisquerydocument.kt
package opentraceability.models.events

class EPCISQueryDocument : EPCISBaseDocument() {
    var QueryName: String = ""
    var SubscriptionID: String = ""

    fun ToEPCISDocument(): EPCISDocument {
        val document = EPCISDocument()

        // get all properties from EPCISBaseDocument
        val props = EPCISBaseDocument::class.java.declaredFields

        // iterate over properties and copy their values to document
        for (p in props) {
            p.isAccessible = true
            val v = p.get(this)
            p.set(document, v)
        }

        return document
    }
}
